tests/unit/cli/test_types.py

```diff
@@ -285,3 +287,32 @@ def test_must_fail_on_invalid_format(self, input):
     def test_successful_parsing(self, input, expected):
         result = self.param_type.convert(input, None, None)
         self.assertEqual(result, expected, msg="Failed with Input = " + str(input))
+
+
+class TestImageRepositoryType(TestCase):
+    def setUp(self):
+        self.param_type = ImageRepositoryType()
+        self.mock_param = Mock(opts=["--image-repository"])
+
+    @parameterized.expand(
+        [
+            # Just a string
+            ("some string"),
```

Comment:
You could maybe add more failing tests, with strings missing key parts of the pattern, like `123456789012.us-east-1.amazonaws.com/test1`